widget/view.controller.js

```diff
@@ -2,38 +2,186 @@
+        var apiPayload = soarConfigService.constants().findKeyStorePayload;
+        apiPayload['filters'][0]['value'] = keyName;
+      }
+      return apiPayload;
+    }
 
-      function moveNext() {
 
-        WizardHandler.wizard('soarFrameworkConfigurationWizard').next();
+    function _handleGblVarsAndKeyStores() {
+      Object.keys($scope.gblVarToKeyStoreMapping).forEach(function (item) {
+        if (item === 'CIDR_Range') {
+          var payload = _buildPayload('sfsp-cidr-range', null, 'findKeyStore');
+          soarConfigService.getKeyStoreRecord(payload, 'keys').then(function (response) {
+            var resp = response['hydra:member'];
+            if (resp.length === 0) {
+              var keyValue = $scope.gblVarToKeyStoreMapping['CIDR_Range'].defaultValue.split(',');
+              var payload = _buildPayload('sfsp-cidr-range', keyValue, 'createKeyStore');
+              soarConfigService.createOrUpdateKeyStore(payload, 'keys').then(function (res) {
+                $scope.defaultGlobalSettings[res.key] = { 'recordValue': res.jSONValue, 'recordUUID': res.uuid };
+              });
+            }
+            else {
+              $scope.defaultGlobalSettings[resp[0].key] = { 'recordValue': resp[0].jSONValue, 'recordUUID': resp[0].uuid };
+            }
+          });
+        }
+        else {
+          soarConfigService.getGBLVariable(item).then(function (response) {
+            // Check if exclude list global variable already present 
+            var resp = response['hydra:member'];
+            if (resp.length > 0) {
+              var gblVarName = resp[0].name;
+              var gblVarID = resp[0].id;
+              var keyName = $scope.gblVarToKeyStoreMapping[gblVarName].keystore;
+              var keyValue = resp[0].value.split(',');
+              var payload = _buildPayload(keyName, keyValue, 'createKeyStore');
+              soarConfigService.createOrUpdateKeyStore(payload, 'keys').then(function (res) {
+                $scope.defaultGlobalSettings[res.key] = { 'recordValue': res.jSONValue, 'recordUUID': res.uuid };
+              });
+              soarConfigService.deleteGBLVariable(gblVarID);
+            }
+            else {
+              var keyName = $scope.gblVarToKeyStoreMapping[item].keystore;
+              var payload = _buildPayload(keyName, null, 'findKeyStore');
+              soarConfigService.getKeyStoreRecord(payload, 'keys').then(function (response) {
+                resp = response['hydra:member'];
+                if (resp.length > 0) {
+                  $scope.defaultGlobalSettings[resp[0].key] = { 'recordValue': resp[0].jSONValue, 'recordUUID': resp[0].uuid };
+                }
+                else {
+                  var keyName = $scope.gblVarToKeyStoreMapping[item].keystore;
+                  var keyValue = $scope.gblVarToKeyStoreMapping[item].defaultValue.split(',');
+                  var payload = _buildPayload(keyName, keyValue, 'createKeyStore');
```

Comment:
keyName, keyValue and payload has already been defined earlier no need to define again, if value is changed just assign it to variable but redefining not required